lib/colors.zsh

```diff
@@ -11,6 +11,10 @@ declare -Ag tw_color_mappings=(
   "info_special" "cyan"
 )
 
+if [[ $(id -u ) -eq 0 ]]; then
+  tw_color_mappings+=( [secondary]=yellow )
+fi
+
```

Comment:
Since this file contains all the logic of color variable creation, I would prefer to leave it managing only that.

What we could do is create variables here for the root user colors, and use them with that condition in `typewritten.zsh`.

I'm thinking of something like:
```shell
# typewritten.zsh
local tw_base_symbol_color="%F{$tw_colors[symbol]}"
if [[ $(id -u ) -eq 0 ]]; then
  tw_base_symbol_color="%F{$tw_colors[symbol_root]}"
fi
```

Then, that variable will be used by the `tw_prompt_color` variable, which I think should be renamed `tw_symbol_color`.

```shell
local tw_symbol_color="%(?,$tw_base_symbol_color,%F{$tw_colors[error_code]})"
```

Let's not forget to use the new variable to reset the symbol color if `$TYPEWRITTEN_DISABLE_RETURN_CODE` is `true`:

```shell
if [ "$TYPEWRITTEN_DISABLE_RETURN_CODE" = true ]; then
  tw_symbol_color="$tw_base_symbol_color"
  tw_return_code=""
fi;
```

Finally, we should use that variable when creating `tw_prompt`:

```shell
tw_prompt="$tw_symbol_color......."
```

These changes need 2 new variables in `lib/colors.zsh` which are `root_modifier` and `symbol_root`:

```shell
declare -Ag tw_color_mappings=(
  ...

  "root_modifier" "yellow"
)

declare -Ag tw_colors=(
  ...
  "symbol_root" $tw_color_mappings[root_modifier]
  ...
)
```

I think that leaves a good separation between the prompt logic and the colors logic. It also has the benefit of letting the user configure the `root` user color. What do you think? Feedback welcome :)

Reply:
Hi @reobin, think you're completely right and the color lib is indeed not the best place. I should have put more thought into this before, I had only looked for the easiest way to implement ... ;)

Let me move this into the main theme, I'll come back with the respective commit.